homeassistant/components/geonetnz_quakes/config_flow.py

```diff
@@ -0,0 +1,89 @@
+"""Config flow to configure the GeoNet NZ Quakes integration."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_RADIUS,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIT_SYSTEM,
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_MINIMUM_MAGNITUDE,
+    CONF_MMI,
+    DEFAULT_MINIMUM_MAGNITUDE,
+    DEFAULT_MMI,
+    DEFAULT_RADIUS,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured GeoNet NZ Quakes instances."""
+    return set(
+        "{0}, {1}".format(entry.data[CONF_LATITUDE], entry.data[CONF_LONGITUDE])
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class GeonetnzQuakesFlowHandler(config_entries.ConfigFlow):
+    """Handle a GeoNet NZ Quakes config flow."""
+
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def _show_form(self, errors=None):
+        """Show the form to the user."""
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_LATITUDE, default=self.hass.config.latitude
+                ): cv.latitude,
+                vol.Optional(
+                    CONF_LONGITUDE, default=self.hass.config.longitude
+                ): cv.longitude,
+                vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS): cv.positive_int,
+                vol.Optional(CONF_MMI, default=DEFAULT_MMI): vol.All(
+                    vol.Coerce(int), vol.Range(min=-1, max=8)
+                ),
+                vol.Optional(
+                    CONF_MINIMUM_MAGNITUDE, default=DEFAULT_MINIMUM_MAGNITUDE
```

Comment:
OK, sorry, I was always under the impression that configuring integrations via the UI was supposed to eventually replace the configuration via YAML.

If we only want critical options in here, I would probably rather keep "MMI", because that is the "indicator of an earthquake's effects on people and their environment", so users likely want to choose this one.
And I would then leave away longitude/latitude, because we're taking the home coordinates by default, and if you want to configure the integration for example for your holiday home, you could go via YAML.
And I would also leave away the minimum magnitude, because that's more of a technical indicator, and probably not as good as the MMI.

Does that sound like a good compromise? Or do you want to keep longitude/latitude as well?

Reply:
Sounds good!

Reply:
OK. Keeping `mmi` and `radius` as the essential options.

Reply:
Is radius essential?

Reply:
My thoughts on `radius` is that this probably has a relationship with `mmi`. For example, if you are only interested in stronger earthquakes you may want to include a larger radius, but if you also want to see weaker earthquakes then a large radius may give you too many results.
I just compared with other integrations - `wwlln` also allows to configure the radius in the form.

Reply:
I just realised that the radius is currently interpreted in km only, not taking the configured unit system into account. I'll fix that now.

Reply:
Fixed handling radius in miles, and added test.